### ISSUE TITLE
www/nginx: support fastcgi backend servers

### ISSUE THREAD

**Issue by darkain** [opnsense/plugins#791]
Right now, it seems back-end servers are automatically used as proxy_pass, but nginx also supports fastcgi_pass. This is extremely useful for PHP web application clusters that expose the PHP fastcgi pool directly, rather than having middleware HTTP servers between nginx and PHP. This should be simple since it is mostly the same as proxy_pass, but just telling nginx to use a different connection protocol to the upstream server.

example config pulled from a working server:
```
        location / {
                include /usr/local/etc/nginx/fastcgi_params;
                fastcgi_connect_timeout 5s;
                fastcgi_param SCRIPT_FILENAME /var/www/index.php;
                fastcgi_pass upstreamcluster;
        }
```

**Comment by fabianfrz** [comment]
Can you post the upstream as well?

**Comment by fabianfrz** [comment]
Local PHP will be in the next release btw: #792

**Comment by darkain** [comment]
This isn't local PHP, this is PHP-FPM running on upstream servers, accessed via FastCGI instead of HTTP. Their upstream configs are the same, only the one directive (proxy vs fastcgi) and associated optional directives change.

```
        upstream fastcgi_upstream {
                hash $remote_addr;    #load balance so each client always connects to same backend server
                server 192.168.1.10:9000;
                server 192.168.1.11:9000;
        }

        upstream http_upstream {
                hash $request_uri;    #load balance connections based on requested file
                server 192.168.1.10:80;
                server 192.168.1.11:80;
        }

        server {
                server_name     example.com

                location /files {
                        proxy_pass http://http_upstream;
                }

                location / {
                        include /usr/local/etc/nginx/fastcgi_params;
                        fastcgi_connect_timeout 5s;
                        fastcgi_param SCRIPT_FILENAME /var/www/index.php;
                        fastcgi_pass fastcgi_upstream;
                }
         }
```

**Comment by fabianfrz** [comment]
Should not be hard to include - I may add it to the PR or add a new one after the current one is merged.

**Comment by fabianfrz** [comment]
solved in c8329fc
